File: src/main/kotlin/me/fungames/jfortniteparse/fort/exports/FortMtxOfferData.java
package me.fungames.jfortniteparse.fort.exports;

import me.fungames.jfortniteparse.fort.objects.FortItemQuantityPair;
import me.fungames.jfortniteparse.ue4.assets.UStruct;
import me.fungames.jfortniteparse.ue4.assets.exports.UPrimaryDataAsset;
import me.fungames.jfortniteparse.ue4.objects.core.i18n.FText;
import me.fungames.jfortniteparse.ue4.objects.core.math.FLinearColor;
import me.fungames.jfortniteparse.ue4.objects.core.math.FVector2D;
import me.fungames.jfortniteparse.ue4.objects.uobject.FPackageIndex;
import me.fungames.jfortniteparse.ue4.objects.uobject.FSoftObjectPath;

import java.util.List;

public class FortMtxOfferData extends UPrimaryDataAsset {
    public FText Header;
    public FText DisplayName;
    public FText ShortDisplayName;
    public FText ShortDescription;
    public FText SubTitleText;
    public FText DisclaimerText;
    public List<FortItemQuantityPair> GrantOverride;
    public FSoftObjectPath SoftTileImage;
    public FVector2D TileImageSize;
    public FSoftObjectPath SoftBadgeImage;
    public FVector2D BadgeImageSize;
    public FSoftObjectPath SoftDetailsImage;
    public FVector2D DetailsImageSize;
    public FSoftObjectPath SoftSimpleImage;
    public FVector2D SimpleImageSize;
    public FPackageIndex /*UMaterialInterface*/ DetailsBadge;
    public List<FortMtxDetailsAttribute> DetailsAttributes;
    public FortMtxGradient Gradient;
    public FLinearColor Background;
    public FSoftObjectPath SoftBackgroundImage;
    public FVector2D BackgroundImageSize;
    public FLinearColor UpsellPrimaryColor;
    public FLinearColor UpsellSecondaryColor;
    public FLinearColor UpsellTextColor;
    public Boolean bUseBaseColors;
    public EFortMtxOfferDisplaySize DisplaySize;
    public List<FortMtxDescriptionAndDetails> DescriptionAndDetailsList;

    @UStruct
    public static class FortMtxDetailsAttribute {
        public FText Name;
        public FText Value;
    }

    @UStruct
    public static class FortMtxGradient {
        public FLinearColor Start;
        public FLinearColor Stop;
    }

    public enum EFortMtxOfferDisplaySize {
        Small,
        Medium,
        Large
    }

    @UStruct
    public static class FortMtxDescriptionAndDetails {
        public FText ShortDescription;
        public List<FortMtxDetailsAttribute> AssociatedDetailsAttributes;
    }
}

File: src/main/kotlin/me/fungames/jfortniteparse/fort/exports/FortMtxOfferData.kt
package me.fungames.jfortniteparse.fort.exports

import me.fungames.jfortniteparse.ue4.assets.UStruct
import me.fungames.jfortniteparse.ue4.assets.exports.UObject
import me.fungames.jfortniteparse.ue4.objects.core.i18n.FText
import me.fungames.jfortniteparse.ue4.objects.core.math.FLinearColor
import me.fungames.jfortniteparse.ue4.objects.slatecore.styling.FSlateBrush
import me.fungames.jfortniteparse.ue4.objects.uobject.FObjectExport

@ExperimentalUnsignedTypes
class FortMtxOfferData : UObject {
    var DisplayName: FText? = null
    var ShortDescription: FText? = null
    var DisclaimerText: FText? = null
    var TileImage: FSlateBrush? = null
    var DetailsImage: FSlateBrush? = null
    var DetailsAttributes: Array<FortMtxDetailsAttribute>? = null
    var Gradient: FortMtxGradient? = null
    var Background: FLinearColor? = null
    var UpsellPrimaryColor: FLinearColor? = null
    var UpsellSecondaryColor: FLinearColor? = null
    var UpsellTextColor: FLinearColor? = null
    var DisplaySize: EFortMtxOfferDisplaySize? = null

    constructor() : super()
    constructor(exportObject: FObjectExport) : super(exportObject)
}

@ExperimentalUnsignedTypes
@UStruct
class FortMtxDetailsAttribute {
    var Name: FText? = null
    var Value: FText? = null
}

@ExperimentalUnsignedTypes
@UStruct
class FortMtxGradient {
    var Start: FLinearColor? = null
    var Stop: FLinearColor? = null
}

enum class EFortMtxOfferDisplaySize { Small, Medium, Large }